### DISCUSSION TITLE
Training semantic role labeling model (with version 3)?

### DISCUSSION THREAD

**Discussion by egumasa** [explosion/spaCy#8447] [Help: Other Questions]
I am finding a way to train a semantic role labeling (and predicate disambiguation) using Ontonote5 dataset. I encountered [this post](https://github.com/explosion/spaCy/discussions/2336#discussion-57793)  from the past, and now that SpaCy v3 has more user-friendly pipelines, I believe that the hurdle is much lower.

To create this training project, which [project template](https://github.com/explosion/projects) would you recommend to start with? I am thinking to leverage the already existing template for either dependency parsing or Ner or entity linking, to start my project. 

I appreciate any recommendation on how to get this project started. I am imagining that training a model with spacy can be efficient than importing AllenNLP models such as [this one](https://demo.allennlp.org/semantic-role-labeling), and [like this](https://gist.github.com/chssch/f788cfd227cb94d0843235a2542026fd).  

I appreciate any suggestions/recommendations! Thank you in advance!

**Comment by egumasa** [comment]
Could anyone provide their insights on how I might do this? OR is this impossible with the current version of spaCy?

**Reply by polm** [reply]
This isn't impossible with the current version of spaCy, but it's also not a built-in feature or something we're currently working on specifically. You could try using the recently released [SpanCategorizer](https://spacy.io/api/spancategorizer) for it, but the main challenge with semantic role labelling is that it's hard to learn in general, and there may be better specialized architectures or approaches.

**Reply by egumasa** [reply]
Thank you for your response on this one! Much appreciated. I see, I think I will stay with the current architecture for now (e.g., importing AllenNLP model into space pipeline). Thanks for the pointer to SpanCategorizer. I think this is interesting new feature! I might consider this for other component I am thinking (e.g., modal verb categorization/sense disambiguation, etc.)

Thank you so much for your guidance!
